### PR TITLE
#116 Requirement Preference page and Diagram display update

### LINKED ISSUE #116: Requirement Preference page and Diagram display update
- The current Requirement Preference page is wrongly formatted and displays incorrect information.
- The Requirement display in diagrams is partially broken, if any model link elements are present in the text field. Furthermore the default display size should be increased.

### PR DESCRIPTION
**Pull request by sandupostaru**
- Update the Requirement Preference page
- Update documentation
- Increase the Requirement content default display size in diagrams
- Fix broken Requirement content display for model link elements

Fixes #116

Change-Id: I7add5d4de1c07ef3e482679178e8f30df640ef86
Signed-off-by: Sandu Postaru <sandu.postaru@thalesgroup.com>

### CODE REVIEW

File: plugins/org.polarsys.capella.vp.requirements.design/src/org/polarsys/capella/vp/requirements/design/service/CapellaRequirementsOpenJavaService.java [hunk 1]
```diff
@@ -295,76 +297,75 @@ public EObject hideRequirements(EObject diagram, Collection<EObject> selectedReq
...
-      expression = queriesAnnotation.getDetails()
-          .get(ReqVPCustomDataHelper.CUSTOM_DATA_KEY_FOR_REQ_VP_QUERIES_CONTENT);
+      expression = queriesAnnotation.getDetails().get(ReqVPCustomDataHelper.CUSTOM_DATA_KEY_FOR_REQ_VP_QUERIES_CONTENT);
       maxLength = queriesAnnotation.getDetails()
           .get(ReqVPCustomDataHelper.CUSTOM_DATA_KEY_FOR_REQ_VP_QUERIES_CONTENT_LENGTH);
     }
-    
-    return LabelHelper.transformHTMLToText(evaluateExpression(session, requirement, expression, maxLength));
+
+    return evaluateExpression(session, requirement, expression, maxLength);
```
Comment (sandupostaru) [inline plugins/org.polarsys.capella.vp.requirements.design/src/org/polarsys/capella/vp/requirements/design/service/CapellaRequirementsOpenJavaService.java:363]: First change

File: plugins/org.polarsys.capella.vp.requirements.design/src/org/polarsys/capella/vp/requirements/design/service/CapellaRequirementsOpenJavaService.java [hunk 2]
```diff
@@ -385,17 +386,21 @@ protected String evaluateExpression(Session session, Requirement requirement, St
           } else {
             resultBuilder.append(value);
           }
-          return reduceString(resultBuilder.toString(), maxLength);
+          String evaluationResult = resultBuilder.toString();
+          String sanytizedResult = LabelHelper.transformHTMLToText(evaluationResult);
+
+          return reduceString(sanytizedResult, maxLength);
```
Comment (sandupostaru) [inline plugins/org.polarsys.capella.vp.requirements.design/src/org/polarsys/capella/vp/requirements/design/service/CapellaRequirementsOpenJavaService.java:392]: Second change

File: plugins/org.polarsys.capella.vp.requirements.ui.commands/src/org/polarsys/capella/vp/requirements/ui/commands/messages.properties [hunk 3]
```diff
@@ -3,8 +3,8 @@ GROUP_TITLE_FOR_CONTENT_AREA=Requirement's content
 EXPRESSION_LABEL_TEXT=Expression
 LENGTH_LABEL_TEXT=Length (put nothing to display full text):
 DefaultValueOfLabelExpression=aql:self.ownedAttributes->select( a | a.definition.ReqIFLongName == 'IE PUID').value
-DefaultValueOfContentExpression=aql:OrderedSet{self.ReqIFLongName, self.ReqIFName, self.ReqIFChapterName, self.ReqIFText}->select(s  | s != 'null' and s.size() > 0).prefix('- ')->sep('\n')
+DefaultValueOfContentExpression=aql:OrderedSet{self.ReqIFLongName, self.ReqIFName, self.ReqIFChapterName, self.ReqIFText}->select(s  | s != 'null' and s.size() > 0).prefix('- ')->sep('\\n')
```
Comment (pdulth) [inline plugins/org.polarsys.capella.vp.requirements.ui.commands/src/org/polarsys/capella/vp/requirements/ui/commands/messages.properties:6]: sure?

**Review by pdulth — APPROVED** [review]

File: plugins/org.polarsys.capella.vp.requirements.ui.commands/src/org/polarsys/capella/vp/requirements/ui/commands/messages.properties (hunk 3 above)
Comment (sandupostaru) [reply plugins/org.polarsys.capella.vp.requirements.ui.commands/src/org/polarsys/capella/vp/requirements/ui/commands/messages.properties:6]: Yes, this was a subtle bug.
Previously the newline separator worked, **as long as the user did not open the 'Requirement Labels in Diagram' dialog**.

When the user opens the dialog, the string is put into a StringFieldEditor and the newline separator is interpreted. Starting from that moment, the newline separator is ignored in the Diagram Display.

Before:
before
image

After:
after
image

**Pull request merged**